src/converters/audio/native_audio_converter.ml

```diff
@@ -50,3 +50,14 @@ let samplerate_converter () =
 
 let () =
   Audio_converter.Samplerate.converters#register "native" samplerate_converter
+
+let channel_layout_converter src dst data =
+  assert (src <> dst);
```

Comment:
Why don't we allow the identity conversion (when `src = dst`)?

Reply:
This case is filtered before so we don't hit the converters for them.